File: server/Infrastructure/Abstractions/Models/ReadModels/EntityListRequest.cs
using Brainvest.Dscribe.Abstractions.Models.Filtering;
using System;
using System.Collections.Generic;
using System.Linq.Expressions;

namespace Brainvest.Dscribe.Abstractions.Models.ReadModels
{
	public class EntityListRequest : IHasEntityTypeName, IOrderRequest, IPageRequest, IFilterModel
	{
		public string EntityTypeName { get; set; }

		public IEnumerable<SortItem> Order { get; set; }
		public int? StartIndex { get; set; }
		public int? Count { get; set; }
		public FilterNodeModel[] Filters { get; set; }
	}

	public class EntityListRequest<TEntity> : IOrderRequest, IPageRequest, IFilterModel<TEntity>
	{
		public IEnumerable<SortItem> Order { get; set; }
		public int? StartIndex { get; set; }
		public int? Count { get; set; }
		public Expression<Func<TEntity, bool>>[] Filters { get; set; }
	}
}

File: client/src/lib/common/models/entity-list-request.ts
import {SortItem} from './sort-item';
import {StorageFilterNode} from '../../filtering/models/storage-filter-node';

export class EntityListRequest {
	constructor(public entityTypeName: string, public filters: StorageFilterNode[],
							public startIndex?: number, public count?: number, public order?: SortItem[]) {
	}

	public getRequestObject(): any {
		return {
			entityTypeName: this.entityTypeName,
			filters: this.filters,
			order: this.order,
			startIndex: this.startIndex,
			count: this.count
		};
	}
}
